application/src/main/java/org/thingsboard/server/service/subscription/TbEntityDataSubCtx.java

```diff
@@ -94,35 +94,36 @@ protected Aggregation getCurrentAggregation() {
 
     private void sendLatestWsMsg(EntityId entityId, String sessionId, TelemetrySubscriptionUpdate subscriptionUpdate, EntityKeyType keyType) {
         Map<String, TsValue> latestUpdate = new HashMap<>();
-        subscriptionUpdate.getData().forEach((k, v) -> {
-            Object[] data = (Object[]) v.get(0);
-            latestUpdate.put(k, new TsValue((Long) data[0], (String) data[1]));
-        });
+        subscriptionUpdate.getData().forEach((key, data) -> data.stream()
+                .filter(o -> o instanceof Object[] && ((Object[]) o).length > 0 && ((Object[]) o)[0] instanceof Long)
+                .max(Comparator.comparingLong(o -> (Long) ((Object[]) o)[0]))
+                .ifPresent(max -> {
+                    Object[] arr = (Object[]) max;
+                    latestUpdate.put(key, new TsValue((Long) arr[0], (String) arr[1]));
+                }));
         EntityData entityData = getDataForEntity(entityId);
```

Comment:
Please make it readable. Perhaps don't use Streams. Extract data[0] and data[1] to properly named variables